Reason: We use a custom name for libperl.so
        (perl)Configure correctly passed that name to the Makefile, (perlcross)configure didn't
Upstream: submitted, https://developer.berlios.de/patch/index.php?func=detailpatch&patch_id=3547&group_id=12304

From 7ad21b9fd78c08ac81591d2397dd974109c11a6b Mon Sep 17 00:00:00 2001
From: Benedikt Morbach <benedikt.morbach@googlemail.com>
Date: Fri, 27 Sep 2013 01:49:44 +0200
Subject: [PATCH] support passing libperl to configure
Organization: Exherbo

---
 Makefile           |  7 +------
 Makefile.config.SH | 21 +++++++++++++++------
 2 files changed, 16 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 58a67ff..2c6a50f 100644
--- a/Makefile
+++ b/Makefile
@@ -161,12 +161,7 @@ static.list: Makefile.config | $(static_tgt) miniperl$X
 
 # ---[ site/library ]-----------------------------------------------------------
 
-libperl$a: op$o perl$o $(obj) $(dynaloader_o)
-	$(AR) cru $@ $(filter %$o,$^)
-	$(RANLIB) $@
-
-libperl$l: op$o perl$o $(obj) $(dynaloader_o) 
-	$(CC) $(LDDLFLAGS) -o $@ $(filter %$o,$^) $(LIBS)
+$(LIBPERL): op$o perl$o $(obj) $(dynaloader_o)
 
 perl.o: git_version.h
 
diff --git a/Makefile.config.SH b/Makefile.config.SH
index 8e80373..a815f1d 100755
--- a/Makefile.config.SH
+++ b/Makefile.config.SH
@@ -130,15 +130,24 @@ perlversion = $perlversion
 dynaloader = $dynaloader
 END
 
+cat >> Makefile.config <<END
+LIBPERL = $libperl
+END
+
 if [ "$useshrplib" == 'define' ]; then
-cat >> Makefile.config <<END
-LIBPERL = libperl\$l
-LIBPERL_LINK = -Wl,-rpath,\$(archlib)/CORE \$(LIBPERL)
+cat >> Makefile.config <<"END"
+LIBPERL_LINK = -Wl,-rpath,$(archlib)/CORE $(LIBPERL)
+
+$(LIBPERL):
+	$(CC) $(LDDLFLAGS) -o $@ $(filter %$o,$^) $(LIBS)
 END
 else
-cat >> Makefile.config <<END
-LIBPERL = libperl\$a
-LIBPERL_LINK = \$(LIBPERL)
+cat >> Makefile.config <<"END"
+LIBPERL_LINK = $(LIBPERL)
+
+$(LIBPERL):
+	$(AR) cru $@ $(filter %$o,$^)
+	$(RANLIB) $@
 END
 fi
 
-- 
1.8.4

